File: leetcode/83.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @return {ListNode}
def delete_duplicates(head)
    output = head
    while (head && head.next)
        if head.val == head.next.val
            head.next = head.next.next
        else
            head = head.next
        end
    end
    output
end

File: leetcode/83.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if not head:
            return head
        prev, cur = head, head
        while cur:
            if cur.val == prev.val:
                cur = cur.next
            else:
                prev.next = cur
                prev = cur
        prev.next = None
        return head
